File: src/test/java/edu/hw1/Task4Test.java
package edu.hw1;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

public class Task4Test {

    Task4 task = new Task4();
    @Test
    @DisplayName("Empty string must return empty string")
    void checkEmptyString(){
        assertThat(task.fixString("")).isEqualTo("");
    }

    @Test
    @DisplayName("String with length 1 must return itself")
    void checkStringWithLengthOne(){
        assertThat(task.fixString("5")).isEqualTo("5");
    }

    @Test
    @DisplayName("6587 must return 5678")
    void check6587(){
        assertThat(task.fixString("6587")).isEqualTo("5678");
    }

    @Test
    @DisplayName("String with odd length should return fixed string on s[:-1]")
    void checkStringWithOddLength(){
        assertThat(task.fixString("21435")).isEqualTo("12345");
    }
}

File: src/test/java/edu/hw3/Task4Test.kt
package edu.hw3

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class Task4Test {

    val task = Task4()

    @Test
    @DisplayName("Method should convert arabic numbers into Roman ones")
    fun check_conversion() {
        val arabic = 3571


        val roman = task.convertToRoman(arabic)


        assertThat(roman).isEqualTo("MMMDLXXI")
    }

    @Test
    @DisplayName("Method should return \"I\" if num is 1 (border case)")
    fun check_one() {
        val arabic = 1;


        val roman = task.convertToRoman(arabic)


        assertThat(roman).isEqualTo("I")
    }

    @Test
    @DisplayName("Method should return \"IV\" if num is 4 (border case)")
    fun check_four() {
        val arabic = 4;


        val roman = task.convertToRoman(arabic)


        assertThat(roman).isEqualTo("IV")
    }

    @Test
    @DisplayName(
        "Method should throw Exception if number not in [1;3999] " +
            "(because is will be not possible to express number only with \"I\", \"V\", \"X\", \"L\", \"C\", \"D\", \"M\")"
    )
    fun check_range() {
        val arabic = 4000


        assertThatThrownBy { task.convertToRoman(arabic) }.isInstanceOf(IllegalArgumentException::class.java)
    }

}
